libs/common/src/lib/search-widget/search-widget-storage.service.ts

```diff
@@ -0,0 +1,157 @@
+    startWith(true),
+    switchMap(() => this.sdk.currentKb.pipe(take(1))),
+    map((kb) => {
+      if (this.standaloneService.standalone) {
+        const configMap: { [kbId: string]: SearchConfiguration[] } = JSON.parse(
+          this.storage.getItem(SEARCH_CONFIGS_KEY) || '{}',
+        );
+        return configMap[kb.id] || [];
+      } else {
+        return (kb.search_configs as SearchAndWidgets)?.searchConfigurations || [];
+      }
+    }),
+  );
+
+  widgetList: Observable<Widget[]> = this.storageUpdated.pipe(
+    startWith(true),
+    switchMap(() => this.sdk.currentKb.pipe(take(1))),
+    map((kb) => {
+      if (this.standaloneService.standalone) {
+        const widgetsMap: { [kbId: string]: Widget[] } = JSON.parse(this.storage.getItem(SAVED_WIDGETS_KEY) || '{}');
+        return widgetsMap[kb.id] || [];
+      } else {
+        return (kb.search_configs as SearchAndWidgets)?.widgets || [];
+      }
+    }),
+    map((widgets) => widgets.sort((a, b) => compareDesc(a.creationDate, b.creationDate))),
+  );
+
+  storeConfigs(updatedConfigs: SearchConfiguration[]) {
+    return this.sdk.currentKb.pipe(
+      take(1),
+      switchMap((kb) => {
+        if (this.standaloneService.standalone) {
+          const configMap: { [kbId: string]: SearchConfiguration[] } = JSON.parse(
+            this.storage.getItem(SEARCH_CONFIGS_KEY) || '{}',
+          );
+          configMap[kb.id] = updatedConfigs;
+          this.storage.setItem(SEARCH_CONFIGS_KEY, JSON.stringify(configMap));
+          return of(undefined);
+        } else {
+          return this.searchAndWidgets.pipe(
+            take(1),
+            switchMap((data) => kb.modify({ search_configs: { ...data, searchConfigurations: updatedConfigs } })),
+            switchMap(() => this.sdk.refreshCurrentKb()),
+          );
+        }
+      }),
+      tap(() => this.storageUpdated.next()),
+    );
+  }
+
+  storeWidgets(updatedWidgets: Widget[]) {
+    return this.sdk.currentKb.pipe(
+      take(1),
+      switchMap((kb) => {
+        if (this.standaloneService.standalone) {
+          const widgetsMap: { [kbId: string]: Widget[] } = JSON.parse(this.storage.getItem(SAVED_WIDGETS_KEY) || '{}');
+          widgetsMap[kb.id] = updatedWidgets;
+          this.storage.setItem(SAVED_WIDGETS_KEY, JSON.stringify(widgetsMap));
+          return of(undefined);
+        } else {
+          return this.searchAndWidgets.pipe(
+            take(1),
+            switchMap((data) => kb.modify({ search_configs: { ...data, widgets: updatedWidgets } })),
+            switchMap(() => this.sdk.refreshCurrentKb()),
+          );
+        }
+      }),
+      tap(() => this.storageUpdated.next()),
+    );
+  }
+
+  migrateConfigsAndWidgets() {
+    if (this.standaloneService.standalone) {
+      return of(undefined);
+    }
+    return this.features.isKbAdmin.pipe(
+      switchMap(() => this.features.isKbAdmin),
+      take(1),
+      filter((isAdmin) => isAdmin),
```

Comment:
I feel like you've got a weird duplicate here
```suggestion
    return this.features.isKbAdmin.pipe(
      take(1),
      filter((isAdmin) => isAdmin),
```